src/app/clusters/scenes-server/scenes-server.cpp

```diff
@@ -0,0 +1,940 @@
+{
+    AddSceneParse<Commands::EnhancedAddScene::DecodableType, Commands::EnhancedAddSceneResponse::Type>(ctx, req, mSceneTable,
+                                                                                                       mGroupProvider);
+}
+
+void ScenesServer::HandleEnhancedViewScene(HandlerContext & ctx, const Commands::EnhancedViewScene::DecodableType & req)
+{
+    ViewSceneParse<Commands::EnhancedViewScene::DecodableType, Commands::EnhancedViewSceneResponse::Type>(ctx, req, mSceneTable,
+                                                                                                          mGroupProvider);
+}
+void ScenesServer::HandleCopyScene(HandlerContext & ctx, const Commands::CopyScene::DecodableType & req)
+{
+    Commands::CopySceneResponse::Type response;
+    CHIP_ERROR err;
+    uint8_t capacity;
+
+    // Response data
+    response.groupIdentifierFrom = req.groupIdentifierFrom;
+    response.sceneIdentifierFrom = req.sceneIdentifierFrom;
+
+    // Verify Endpoint in group
+    VerifyOrReturn(nullptr != mGroupProvider);
+    if ((!mGroupProvider->HasEndpoint(ctx.mCommandHandler.GetAccessingFabricIndex(), req.groupIdentifierFrom, mPath.mEndpointId) &&
+         req.groupIdentifierFrom != 0) ||
+        (!mGroupProvider->HasEndpoint(ctx.mCommandHandler.GetAccessingFabricIndex(), req.groupIdentifierTo, mPath.mEndpointId) &&
+         req.groupIdentifierTo != 0))
+    {
+        response.status = static_cast<uint8_t>(Protocols::InteractionModel::Status::InvalidCommand);
+        ctx.mCommandHandler.AddResponse(mPath, response);
+        return;
+    }
+
+    // Get Capacity
+    err = mSceneTable->GetRemainingCapacity(ctx.mCommandHandler.GetAccessingFabricIndex(), capacity);
+    if (err != CHIP_NO_ERROR)
+    {
+        response.status = static_cast<uint8_t>(Protocols::InteractionModel::Status::Failure);
+        ctx.mCommandHandler.AddResponse(mPath, response);
+        return;
+    }
+
+    // Checks if we copy a single scene or all of them
+    if (req.mode.GetField(app::Clusters::Scenes::ScenesCopyMode::kCopyAllScenes))
+    {
+        // Scene Table interface data
+        SceneId scenesInGroup[scenes::kMaxScenesPerFabric];
+        Span<SceneId> sceneList = Span<SceneId>(scenesInGroup);
+
+        // populate scene list
+        err = mSceneTable->GetAllSceneIdsInGroup(ctx.mCommandHandler.GetAccessingFabricIndex(), req.groupIdentifierFrom, sceneList);
+        if (err != CHIP_NO_ERROR)
+        {
+            response.status = static_cast<uint8_t>(Protocols::InteractionModel::Status::Failure);
+            ctx.mCommandHandler.AddResponse(mPath, response);
+            return;
+        }
+
+        if (capacity == 0)
+        {
+            response.status = static_cast<uint8_t>(Protocols::InteractionModel::Status::ResourceExhausted);
+            ctx.mCommandHandler.AddResponse(mPath, response);
+            return;
+        }
+
+        for (auto & SceneId : sceneList)
```

Comment:
Since the spec states: 

> If, during the copy, there is no more space in the Scene Table for the entire next scene to be copied, the Status field SHALL be set to RESOURCE_EXHAUSTED and the scenes already copied SHALL be kept.

I assumed that I would copy scenes until `SetSceneTableEntry` fails due to lack of space, the generating RESOURCE_EXHAUSTED.

Reply:
That seems fine, yes, until spec gets sorted.